kahuna/public/js/util/storage.js

```diff
@@ -0,0 +1,24 @@
+import angular from 'angular';
+
+export const storage  = angular.module('util.storage', []);
+
+storage.factory('storage', ['$window', function($window) {
+    function setItem(key, val) {
+        const storeVal = typeof val === 'string' ? val : JSON.stringify(val);
+        $window.localStorage.setItem(key, storeVal);
+    }
+
+    function getItem(key) {
+        let val = $window.localStorage.getItem(key);
+        try {
+            val = JSON.parse(val);
+        } catch (_) {}
```

Comment:
If you really want to allow any type, then it would be nice to enforce that the type you put in is the type you get out. Could do this different ways I guess (store type hint alongside the value, different `getItem` methods to retrieve different types or type argument to `getItem`, etc).

It's all a little bit superstitious, but I find the simpler the API (you know what you put in and get out), the less things tend to go wrong. In practice it doesn't seem like you're using this API to save strings now, so why generalise it to strings?